### ISSUE TITLE
error when using the "http request" node, with the uploaded file larger than 900KB

### ISSUE THREAD

**Issue by Luminous-seed** [langgenius/dify#12216]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.14.2

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

# Problem 
It seems that when dify's HTTP request uploads a file, with the file larger than about 800KB, it will not be able to send the file successfully.
I have already modified several parameters, such as:

`MAX_VARIABLE_SIZE=2048000   not 204800`,
`WORKFLOW_FILE_UPLOAD_LIMIT=50 not 10`,
```
# HTTP request node in workflow configuration
HTTP_REQUEST_NODE_MAX_BINARY_SIZE=10485760000 not 10485760
HTTP_REQUEST_NODE_MAX_TEXT_SIZE=1048576000 not 1048576
```
` NGINX_CLIENT_MAX_BODY_SIZE=200M not 15M`
 
However, These modifications are not helpful

# preparation
I use the workflow to deal with pdf, whose first step is OCR. 
I have completed the OCR backend and all services run normally.
I tested the OCR backend service with some files with size from 100KB to 25MB. Some records are shown below.

SUCCESS with 927KB
![image](https://github.com/user-attachments/assets/d548daa6-6aa8-4227-bfe8-0c71536a0955)

SUCCESS with myfilename_test.pdf 5566KB
![image](https://github.com/user-attachments/assets/4aa6a5fa-d64e-4e5b-939d-2e908fd5263f)

# work in DIFY
BUT In DIFY
1. the file is  small, like 136KB.
the success requests is like this:
![image](https://github.com/user-attachments/assets/70dbfc37-3dd3-4431-9827-1856af9c2556)

2. the file is  larger, like 900KB. ERROR OCCURS.
but when the file is larger than 900KB, the error happened, like Picture ERROR01

Picture ERROR01 (927KB)
![image](https://github.com/user-attachments/assets/98447f46-8df3-44ae-9857-6a472afdb8f4)

but when the file is 5566KB, the error happened, like Picture ERROR02

Picture ERROR02 (5566KB)
![image](https://github.com/user-attachments/assets/10e0c7ac-1917-4b2b-ab72-69e93d4aeb3c)

The errors log is like ：
``` 
10.14.121.217 - - [30/Dec/2024 16:18:53] "POST /local_ocr HTTP/1.1" 400 -
10.14.121.217 - - [30/Dec/2024 16:18:53] "POST /local_ocr HTTP/1.1" 400 -
```
and:
I debug the backend with the files for Picture ERROR01 and Picture ERROR02.  The errors occur at :
```
@app.route('/local_ocr', methods=['POST'])
def local_ocr():

    file_data = request.get_data()
    if not file_data:
        flash('No file data provided')
        return redirect(request.url)
        ......
```
error occurs at `file_data = request.get_data()`,with `request.get_data()` log:
```
# request.get_data()
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/skl_code/anaconda3/envs/minerU/lib/python3.10/site-packages/werkzeug/wrappers/request.py", line 419, in get_data
    rv = self.stream.read()
  File "/home/skl_code/anaconda3/envs/minerU/lib/python3.10/site-packages/werkzeug/wsgi.py", line 577, in readall
    data = self.read(1024 * 64)
  File "/home/skl_code/anaconda3/envs/minerU/lib/python3.10/site-packages/werkzeug/wsgi.py", line 562, in readinto
    self.on_disconnect()
  File "/home/skl_code/anaconda3/envs/minerU/lib/python3.10/site-packages/werkzeug/wsgi.py", line 499, in on_disconnect
    raise ClientDisconnected()
werkzeug.exceptions.ClientDisconnected: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.
```
The log indicates that the backend did not receive the file at all！！！

### ✔️ Expected Behavior

the https request should be successful with files larger than 800KB

### ❌ Actual Behavior

the https request failed with `Server disconnected without sending a response.`  Because the server did not receive the correct file at all. I can conclude that this is a problem with dify, because the same https request can be successful when the file is small.

**Comment by Luminous-seed** [comment]
But I have changed the `NGINX_CLIENT_MAX_BODY_SIZE=200M` and `# Upload image file size limit, default 10M.
UPLOAD_IMAGE_FILE_SIZE_LIMIT=100
# Upload video file size limit, default 100M.
UPLOAD_VIDEO_FILE_SIZE_LIMIT=1000
# Upload audio file size limit, default 50M.
UPLOAD_AUDIO_FILE_SIZE_LIMIT=500`

**Comment by Luminous-seed** [comment]
@dosu  I have changed the `NGINX_CLIENT_MAX_BODY_SIZE=200M` and `UPLOAD_FILE_SIZE_LIMIT= 150MB` in `.env` and run 
```
docker-compose down
docker-compose up --build -d
```

But nothing changed. This suggestion doesn't seem to work.

**Comment by Luminous-seed** [comment]
@dosu I check the env in container `docker-nginx-1` ,the log is shown below:
```
user@Ubuntu:/dify/docker$ docker exec -it docker-nginx-1 env
...Omit some unimportant parts...
HOSTNAME=0282579a2c57
TERM=xterm
NGINX_PORT=80
NGINX_SSL_PORT=443
NGINX_PROXY_READ_TIMEOUT=3600s
NGINX_SSL_CERT_KEY_FILENAME=dify.key
NGINX_ENABLE_CERTBOT_CHALLENGE=false
NGINX_PROXY_SEND_TIMEOUT=3600s
NGINX_CLIENT_MAX_BODY_SIZE=200M
```
It is clear that the `NGINX_CLIENT_MAX_BODY_SIZE` has been changed from 50MB to 200MB
but the error I encoutered with larger file still exists.( the file here is a new file that could be successfully with postman request, but failed with dify's https request )
![image](https://github.com/user-attachments/assets/8285b4f8-c65f-490d-afc3-87ea409fb19d)

and when the file is smaller, it(a new file with size at 526KB) can be successful
![image](https://github.com/user-attachments/assets/4759dd22-612a-463f-a897-21ef20005d1a)

So in conclusion, nothing happen. This suggestion doesn't seem to work. 
Also, as you mentioned, there may be restrictions on the size of uploaded files in the 'network and proxy' or other parts of the code, which is exactly what I am looking for. If possible, please provide more assistance.

**Comment by Luminous-seed** [comment]
@dosu Here is more information:
For the 1: I attached shell with container `docker-api-1 `,obtaining the log.
```
root@9358d17301f6:/app/api# echo $HTTP_REQUEST_NODE_MAX_BINARY_SIZE 
10485760000
```
`HTTP_REQUEST_NODE_MAX_BINARY_SIZE = 10485760000` is larager for a file with size at 5MB.

For the 2:
I set the limitation in `docker/.env` like belew, and run `docker compose up -d` ,so I think the the content of `web/app/components/base/file-uploader/constants.ts` is not effective and has no impact on the current error message.
```
# Upload file size limit, default 15M.
UPLOAD_FILE_SIZE_LIMIT=250
```

for the 3:
I attached shell with container `docker-api-1 `,obtaining the log.
```
root@9358d17301f6:/app/api# echo $UPLOAD_FILE_SIZE_LIMIT
250
```
and in `api/services/file_service.py`, with code:
```
    def is_file_size_within_limit(*, extension: str, file_size: int) -> bool:
        if extension in IMAGE_EXTENSIONS:
            file_size_limit = dify_config.UPLOAD_IMAGE_FILE_SIZE_LIMIT * 1024 * 1024
        elif extension in VIDEO_EXTENSIONS:
            file_size_limit = dify_config.UPLOAD_VIDEO_FILE_SIZE_LIMIT * 1024 * 1024
        elif extension in AUDIO_EXTENSIONS:
            file_size_limit = dify_config.UPLOAD_AUDIO_FILE_SIZE_LIMIT * 1024 * 1024
        else:
            file_size_limit = dify_config.UPLOAD_FILE_SIZE_LIMIT * 1024 * 1024
```
A 5MB or 927KB PDF file will not exceed the limit. So I don't think the problem lies here

for the 4:
maybe what you said could be right , but the fact is the process fail when the file is with size at 927KB. I upload the file in LAN. I don't think the problem lies here, because the process of uploading files won't take more than five seconds in my opinion, 

for the 5: I don't know how to check. But I don't think the problem lies here.

**Comment by azhou125** [comment]
@Luminous-seed Did you solve the problem? I encountered the same issue.

**Comment by Luminous-seed** [comment]
After conducting so many tests, I believe that the design of the HTTPS request node is a problem with this issue(maybe my opinion is wrong, but faor me, I cannot solve this problem ). So I changed my strategy. HTTPS requests themselves can be completed through Python code, so I created a code node to accomplish the task.

If you want something about my ecxperience, you can contact me,  because some of the code involves privacy.

**Comment by Luminous-seed** [comment]
> @Luminous-seed Did you solve the problem? I encountered the same issue.

After conducting so many tests, I believe that the design of the HTTPS request node is a problem with this issue(maybe my opinion is wrong, but faor me, I cannot solve this problem ). So I changed my strategy. HTTPS requests themselves can be completed through Python code, so I created a code node to accomplish the task.

If you want something about my ecxperience, you can contact me, because some of the code involves privacy.

**Comment by azhou125** [comment]
> > @Luminous-seed Did you solve the problem? I encountered the same issue.
> 
> After conducting so many tests, I believe that the design of the HTTPS request node is a problem with this issue(maybe my opinion is wrong, but faor me, I cannot solve this problem ). So I changed my strategy. HTTPS requests themselves can be completed through Python code, so I created a code node to accomplish the task.
> 
> If you want something about my ecxperience, you can contact me, because some of the code involves privacy.

Absolutely. How did you pass file variable to python code node? In my version of dify, "Code" node doesn't support passing "File" object  as an argument in python function.

**Comment by Luminous-seed** [comment]
@azhou125 Here is the key process. 

![image](https://github.com/user-attachments/assets/b248440a-5b81-4ffe-924e-6ceb828a40ea)

**Comment by Luminous-seed** [comment]
@azhou125  Here is the key process.
![image](https://github.com/user-attachments/assets/e7f0eddc-851c-4ef7-8d51-5dfba467d02d)

**Comment by azhou125** [comment]
@Luminous-seed As far as I know, this "File->url" variable just pass an empty string "" in version 14.2 (and 15.0).

Is that the same case on your side?

**Comment by liliang1234** [comment]
@Luminous-seed Did you solve the problem? I encountered the same issue.

**Comment by liliang1234** [comment]
> @Luminous-seed As far as I know, this "File->url" variable just pass an empty string "" in version 14.2 (and 15.0).
> 
> Is that the same case on your side?

I have the same problem.

**Comment by Luminous-seed** [comment]
@azhou125 @liliang1234  
I also failed to run this module today because the 'File ->URL' variable is empty. Specifically, when a file is uploaded, the 'File ->URL' variable itself has a value, but it cannot be passed during transmission. 
I also have this situation now. But when I tested a few days ago, it ran normally and the variable was passed correctly. I don't know   how to fix. @laipz8200

**Comment by laipz8200** [comment]
> [@azhou125](https://github.com/azhou125) [@liliang1234](https://github.com/liliang1234) I also failed to run this module today because the 'File ->URL' variable is empty. Specifically, when a file is uploaded, the 'File ->URL' variable itself has a value, but it cannot be passed during transmission. I also have this situation now. But when I tested a few days ago, it ran normally and the variable was passed correctly. I don't know how to fix. [@laipz8200](https://github.com/laipz8200)

You can reference this discussion: #10687.

**Comment by h4ckdepy** [comment]
Hello, could you provide a desensitized DSL file? @Luminous-seed

**Comment by chilejiang1024** [comment]
Is this issue fixed? I found that it is still not working when uploading files bigger than 1MB in http request node. The version is 0.15.3.

![Image](https://github.com/user-attachments/assets/47435414-8b35-48cd-af71-911ce0fdcf77)
